### ISSUE TITLE
Wrong tax amount calculation on virtual products when using Express Checkouts (PRB and ECE)

### ISSUE THREAD

**Issue by asumaran** [Automattic/woocommerce-payments#9020]
### Describe the bug
Tax amount presented to the user is incorrect for virtual products when using express checkout buttons, both PRB and ECE.

It seems to be defaulting to the store's country instead of taking the customer's billing address.

### To Reproduce

Assuming you have a working site with WooPayments **in test mode to avoid charging your card** and Express Checkouts enabled:

- Make sure the option "Enable tax rates and calculations" is enabled in WooCommerce > Settings.
- Go to WooCommerce > Tax and add two standard tax rates.
  - Country code: `PE`, Rate %:  `35.0000`, Tax name: `PETAX` – (**note for reviewer: Use your country's code as long it's not the US**)
  - Country code: `US`, Rate %:  `11.0000`, Tax name: `USATAX`
- Create a simple virtual product. Set the price to `42`
- Go to the product page created in the previous step.
  - Delete all cookies. (Go to the Chrome Dev tools > Application > Cookies > Click "Clear all cookies" button).
  - Then refresh the product page.
- Click the "Buy with GPay" button.

### Actual behavior

Notice the total amount is `46.62`. Your selected card has to have an address for the country for which you created the tax rate.

This is incorrect because the total amount is calculated using the store's country (US): `42 + 42 * 0.11 = 46.62` which is wrong it should use the customer's country.

### Screenshots

<img width="599" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/1025173/24e458e9-9206-4b06-841b-daa0ea77a38a">

### Expected behavior

The total amount should be: `42 + 42 * 0.35 = 56.7`

If you finish the checkout process by clicking "Pay" you'll be charged with the correct amount which is different to what was presented before in the payment sheet.

**Comment by c-shultz** [comment]
@frosso do you think the use of tokenized carts will address this? I know that we're sidestepping a lot of WC core functionality with our existing express checkout integrations.

**Comment by asumaran** [comment]
@c-shultz, the problem is that we obtain the billing address after the user submits the payment sheet, and we need it before opening it to calculate and show the tax amount. It defaults to the store's country because it is set up by default.

<img width="671" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/1025173/5a464123-31c5-4afb-a37f-bbb1c43a17f2">

Sometimes, the tax amount is calculated correctly if the customer's billing address matches the store's country, but this will not always be the case. I'm located in a different country than the store I'm testing, and I often see a mismatch between what was presented and what was charged.

**Comment by frosso** [comment]
> @frosso do you think the use of tokenized carts will address this? I know that we're sidestepping a lot of WC core functionality with our existing express checkout integrations.

@c-shultz tokenized cart won't resolve this. This ticket seems to be a duplicate of https://github.com/Automattic/woocommerce-payments/issues/8373 - I added a comment here about my findings with tokenized cart: https://github.com/Automattic/woocommerce-payments/issues/8373#issuecomment-2182982856

**Comment by asumaran** [comment]
@frosso, it's the same issue. You and @gpressutto5 in #8373 have already highlighted all the issues I found myself.

The main problem is that we can't know the billing address in advance to calculate the tax amount accurately. There's no API available to retrieve the billing address associated with the payment method. Even if there were a way to implement that, we are constrained by the 1-second limitation before calling the `resolve` function.

When I tested with subscriptions, I encountered the same bug even when the user was logged in. Sometimes, the tax amount was correct by coincidence, but this isn't reliable as not always the saved billing address will match the billing address attached to the payment method.

I discussed this with @bborman22, and he suggested obtaining the billing address during the 'confirm' event before processing the order. If the billing address and tax amount don't match the amount shown to the shopper, we would abort the payment and reinitialize the ECE/PRB button with the updated billing address. While this approach might work, it's not an ideal user experience. The downside of doing this approach is that if the shopper uses a payment method with a different country we'll abort the payment again.

I'm closing this issue in favor of #8373
